lib/active_merchant/billing/gateways/payway_dot_com.rb

```diff
@@ -0,0 +1,268 @@
+        post[:cardTransaction][:eciType] = eci_type
+        # need source, required or will return source not found
+        post[:cardTransaction][:idSource] = options[:source_id] if options[:source_id]
+        # optional processorSoftDescriptor
+        post[:cardTransaction][:processorSoftDescriptor] = options[:processor_soft_descriptor] if options[:processor_soft_descriptor]
+        # optional tax amount
+        post[:cardTransaction][:tax] = options[:tax] if options[:tax]
+      end
+
+      def add_card_payment(post, payment, options)
+        # credit_card
+        post[:accountInputMode] = "primaryAccountNumber"
+
+        post[:cardAccount] ||= {}
+        post[:cardAccount][:accountNumber] = payment.number
+        post[:cardAccount][:fsv]           = payment.verification_value
+        post[:cardAccount][:expirationDate]    = expdate(payment)
+        # optional data
+        post[:cardAccount][:email]     = options[:email]
+      end
+
+      def expdate(credit_card)
+        year  = format(credit_card.year, :four_digits)
+        month = format(credit_card.month, :two_digits)
+
+        # return MMYYYY
+        month + year
+      end
+
+      def parse(body)
+        body.blank? ? {} : JSON.parse(body)
+      end
+
+      def commit(action, parameters)
+        # set request name
+        parameters[:request] = action
+
+        url = (test? ? test_url : live_url)
+        payload = parameters.to_json unless parameters.nil?
+
+        response =
+          begin
+            parse(ssl_request(:post, url, payload, headers ))
+            rescue ResponseError => e
+            return Response.new(false, 'Invalid Login') if e.response.code == '401'
+
+            parse(e.response.body)
+          end
+
+        success = success_from(response)
+        avs_result_code = response['cardTransaction'].nil? || response['cardTransaction']['addressVerificationResults'].nil? ? "" : response['cardTransaction']['addressVerificationResults']
+        avs_result = AVSResult.new(code: AVS_MAPPING[avs_result_code])
+        cvv_result = CVVResult.new(response['cardTransaction']['fraudSecurityResults']) if response['cardTransaction'] && response['cardTransaction']['fraudSecurityResults']
+
+        Response.new(
+          success,
+          message_from(success, response),
+          response,
+          test: test?,
+          error_code: error_code_from(response),
+          authorization: authorization_from(response),
+          avs_result: avs_result,
+          cvv_result: cvv_result
+        )
+      end
+
+      def success_from(response)
+        response['paywayCode'] == PAYWAY_WS_SUCCESS
+      end
+
+      def error_code_from(response)
+        if success_from(response)
+          ''
+        else
+          error = !STANDARD_ERROR_CODE_MAPPING[response['paywayCode']].nil? ?
+          STANDARD_ERROR_CODE_MAPPING[response['paywayCode']] :
+          STANDARD_ERROR_CODE[:processing_error]
+        end
+      end
```

Comment:
Is there a reason to set error_code to empty string versus just keep it nil? Anyway I think the method can be simplified to something like 
```suggestion
      def error_code_from(response)
        return '' if success_from(response)
        
        error = !STANDARD_ERROR_CODE_MAPPING[response['paywayCode']].nil? ? 
        STANDARD_ERROR_CODE_MAPPING[response['paywayCode']] : 
        STANDARD_ERROR_CODE[:processing_error]
        return error
      end
```

Or can just do `return if success_from(response)` if there is no specific reason to set error to empty string